Sources/DistributedActors/ClusterSystem.swift

```diff
@@ -1019,20 +1029,98 @@ public struct ClusterInvocationResultHandler: DistributedTargetInvocationResultH
     }
 
     public func onReturn<Success: Codable>(value: Success) async throws {
-        let ref = _ActorRef<Success>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
-        ref.tell(value)
+        let ref = _ActorRef<RemoteCallReply<Success>>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
+        ref.tell(.init(value: value))
     }
 
     public func onReturnVoid() async throws {
-        let ref = _ActorRef<_Done>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
-        ref.tell(_Done.done)
+        let ref = _ActorRef<RemoteCallReply<_Done>>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
+        ref.tell(.init(value: _Done.done))
     }
 
     public func onThrow<Err: Error>(error: Err) async throws {
         self.system.log.warning("Result handler, onThrow: \(error)")
-        // FIXME(distributed): carry the error back
-        fatalError("FIXME: implement sending back errors")
+        let ref = _ActorRef<RemoteCallReply<_Done>>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
+        if let codableError = error as? (Error & Codable) {
+            ref.tell(.init(error: codableError))
+        } else {
+            ref.tell(.init(error: GenericRemoteCallError(message: "Remote call error of [\(type(of: error as Any))] type occurred")))
+        }
+    }
```

Comment:
Awesome 👍 

I think we may want to indeed do some configuration here... some "default remote call error handling" and set it to `.genericRemoteCallError` or `.opaqueError` if we'd want to not expose the exact error that was thrown but just "some error happened" I guess?

That's a separate ticket though and not for beta1 :nod:

Reply:
Created https://github.com/apple/swift-distributed-actors/issues/932